kernel/kernel-api/src/main/java/io/delta/kernel/Table.java

```diff
@@ -57,6 +58,10 @@ static Table forPath(Engine engine, String path) {
     return TableImpl.forPath(engine, path);
   }
 
+  static Table forPathWithTableId(Engine engine, String path, TableIdentifier tableId) {
```

Comment:
This is going to show up the javadocs. Add the docs.

Alternative: did you consider the `forPath(Engine, String path, Optional<TableIdentifier> tableId)`. It will break the current connectors, but not a major issue to change it. This API just going to repeat the most of the same javadoc above.

To simplify may use `@see #forPath` in javadocs.

Reply:
I did consider changing the `forPath` API. I'm not included to break the existing API. WDYT?

cc @nicklan

Reply:
It will be a breaking change, but the connectors will not know there is an option they can use. Later on when we add CC, they will be forced to use the other API and it is not clear how connector upgrade to use the new table.forpath API works.

Reply:
anyways, it is just a suggestion, not blocking.

Reply:
> Later on when we add CC, they will be forced to use the other API

Including a table identifier is not required for CC, btw